### PR TITLE
ENH/BUG: infer reso in array_strptime

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/tslibs/test_strptime.py [hunk 1]
```diff
@@ -59,3 +59,39 @@ def test_array_strptime_resolution_mixed(self, tz):
...
+        res2, _ = array_strptime(
+            vals[::-1], fmt="%Y-%m-%d", utc=False, creso=creso_infer
+        )
+
+        # 1ms is an arbitrary cutoff for call overhead; in local testing the
+        #  actual difference is about 250us
+        tolerance = np.timedelta64(1, "ms")
+
+        assert res.dtype == "M8[us]"
+        assert abs(res[0] - now) < tolerance
```
Comment (mroeschke) [inline pandas/tests/tslibs/test_strptime.py:78]: I would be OK skipping the comparison to a separate "now" to potentially avoid near-miss occasional failures. I think at least passing `"today"` and `"now"` through `array_strptime` and seeing if `us` is return is good enough

Comment (jbrockmendel) [reply pandas/tests/tslibs/test_strptime.py:78]: would you be happier with a bigger tolerance?  I can imagine regressions here that cause this to miss by a huge amount, hard to imagine regressions that miss by a small amount

Comment (mroeschke) [reply pandas/tests/tslibs/test_strptime.py:78]: Yeah a larger tolerance would give me more peace.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**